reactor-core/src/main/java/reactor/core/publisher/SinksSpecs.java

```diff
@@ -125,12 +128,26 @@ public void emitNext(T value) {
 				emitError(ex);
 				Operators.onDiscard(value, currentContext());
 				break;
+			}
 			case FAIL_CANCELLED:
 				Operators.onDiscard(value, currentContext());
 				break;
 			case FAIL_TERMINATED:
 				Operators.onNextDropped(value, currentContext());
 				break;
+			case FAIL_NON_SERIALIZED: {
```

Comment:
could we maybe loop (maybe with a `parkNanos(10)` or something) in order to at least ensure that `emitXxx` methods don't require external synchronization?

Reply:
Unfortunately we can't, otherwise we may introduce an infinite loop. We could retry it once, but IMO if retries are desired, `tryEmitXxx` should be used instead.